### PR TITLE
Set up GitHub CI for rustsbi/xuantie with formatting and RISC-V checks

### PR DESCRIPTION
**Pull request by chenzongyao200127**
- Add `cargo fmt` to enforce code style.
- Run `cargo check` on the `xuantie-riscv` library for multiple RISC-V targets.

Signed-off-by: Zongyao Chen solar1s@163.com

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: Xuantie RISC-V CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main, master]
```
Comment (luojia65) [inline .github/workflows/ci.yml]:
```suggestion
    branches: [main]
```

Comment (luojia65) [reply .github/workflows/ci.yml]: There's only `main` branch in this project.

**Review by luojia65 — APPROVED** [review]
The code generally looks OK to me.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (chenzongyao200127) [reply .github/workflows/ci.yml]: Thank you for pointing that out!

**Pull request merged**